micro_dl/input/tile_stack.py

```diff
@@ -16,46 +15,89 @@ class ImageStackTiler:
     def __init__(self,
                  input_dir,
                  output_dir,
-                 tile_size,
-                 step_size,
+                 tile_dict,
+                 tile_size=[256, 256],
+                 step_size=[64, 64],
+                 depth=1,
                  time_ids=-1,
                  channel_ids=-1,
                  slice_ids=-1,
+                 pos_ids=-1,
                  hist_clip_limits=None,
                  flat_field_dir=None,
-                 isotropic=False):
+                 isotropic=False,
+                 data_format='channels_first'):
         """
         Normalizes images using z-score, then tiles them.
         Isotropic here refers to the same dimension/shape along row, col, slice
         and not really isotropic resolution in mm.
 
         :param str input_dir: Directory with frames to be tiled
         :param str output_dir: Base output directory
-        :param list/tuple/np array tile_size: size of the blocks to be cropped
+        :param list tile_size: size of the blocks to be cropped
          from the image
-        :param list/tuple/np array step_size: size of the window shift. In case
+        :param list step_size: size of the window shift. In case
          of no overlap, the step size is tile_size. If overlap, step_size <
          tile_size
+        :param int depth: Determines z depth for generating stack training data
+            Default 1 assumes 2D data
         :param list/int time_ids: Tile given timepoint indices
         :param list/int tile_channels: Tile images in the given channel indices
          default=-1, tile all channels
         :param int slice_ids: Index of which focal plane acquisition to
          use (for 2D). default=-1 for the whole z-stack
+        :param int pos_ids: Position (FOV) indices to use
         :param list hist_clip_limits: lower and upper percentiles used for
          histogram clipping.
         :param str flat_field_dir: Flatfield directory. None if no flatfield
             correction
         :param bool isotropic: if 3D, make the grid/shape isotropic
+        :param str data_format: Channels first or last
         """
         self.input_dir = input_dir
         self.output_dir = output_dir
+
+        if 'depth' in tile_dict:
+            depth = tile_dict['depth']
+        assert depth in {1, 3, 5},\
```

Comment:
wondering if we need this assert here in case we want to use/extend this for 3D images as well!